### PR TITLE
Seprate networking

### PR DESCRIPTION
**Pull request by avrham408**

### CODE REVIEW

File: torrent/io.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+def create_file(path, size):
+    f = open(path, 'wb')
+    f.seek(size - 1)
+    f.write(b'\0')
+    f.close()
+
+
+def get_path(*args, base=MODULE_PATH):
+    # if you want create_path that not connect to module path just path=False
```
Comment (adarmaor) [inline torrent/io.py]: this comment is irrelevant right?

File: torrent/io.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    if base:
+        return os.path.join(base, *args)
+    else:
+        return os.path.join(*args) 
+def copy_file(src, dest):
```
Comment (adarmaor) [inline torrent/io.py:39]: spaces before 🙈

File: torrent/peer_protocol.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+   The file contain the implmentaion in oop of every class and data that belong to him
+"""
+import logging
+import struct
+from bitarray import bitarray
+
+logger = logging.getLogger(__name__)
+
+
+class Message():
```
Comment (adarmaor) [inline torrent/peer_protocol.py:22]: what is this?

File: torrent/io.py (hunk 1 above)
Comment (avrham408) [reply torrent/io.py]: why?
if in the future i want to add support for another download dir,
i want the function could support this.
it's not good idea?

File: torrent/peer_protocol.py (hunk 3 above)
Comment (adarmaor) [reply torrent/peer_protocol.py:22]: why are all the message_id's None?

File: torrent/peer_protocol.py [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+        except struct.error:
+            logger.debug("parse info hash failed")
+            return None
+        return cls(info_hash, peer_id)
+
+
+class Keep_Alive(Message):
+    message_id = None
+
+    def __init__(self):
```
Comment (adarmaor) [inline torrent/peer_protocol.py]: redundant

File: torrent/peer_protocol.py (hunk 3 above)
Comment (adarmaor) [reply torrent/peer_protocol.py:22]: ah i see

File: torrent/peer_protocol.py (hunk 4 above)
Comment (adarmaor) [reply torrent/peer_protocol.py]: all the inits that do nothing can be removed

File: torrent/io.py (hunk 2 above)
Comment (avrham408) [reply torrent/io.py:39]: fixing all the pep 8 now.
we wanted to create this PR because i sent my CV.

File: torrent/tracker.py [hunk 5]
```diff
@@ -77,8 +77,8 @@ def udp_request(tracker, torrent_file, peer_manager, wait=0, recursive=False):
         interval, peers = announce_response
         parsed_peers = parse_peers(peers)
         peer_manager.add_peers(parsed_peers)
-        logger.info(f"{len(peers)}peers add to peer_manager and go to sleep for {interval} seconds")
-        sleep(interval)
+        logger.info(f" {len(peers)} peers add to peer_manager and go to sleep for {interval} seconds")
+        sleep(int(interval / 5))
```
Comment (adarmaor) [inline torrent/tracker.py]: is interval a float or string?

File: torrent/io.py (hunk 1 above)
Comment (adarmaor) [reply torrent/io.py]: the base param is good, but what is path=False?

Comment (adarmaor) [reply torrent/io.py]: did you mean `base= False`

File: torrent/tracker.py (hunk 5 above)
Comment (adarmaor) [reply torrent/tracker.py]: and this?

Comment (adarmaor) [reply torrent/tracker.py]: like, why do you need int

**Review by adarmaor — APPROVED** [review]

**Pull request merged**